src/configuration.js

```diff
@@ -79,47 +82,75 @@ export class Configuration {
   }
 
   getRules() {
-    var rules = defaultRules;
+    var rules = this.getAllRules();
     var specifiedRules;
 
     if (this.options.rules && this.options.rules.length > 0) {
       specifiedRules = this.options.rules.map(toUpperCamelCase);
-      rules = rules.filter(rule => {
+      rules = this.getAllRules().filter(rule => {
         return specifiedRules.indexOf(rule.name) >= 0;
       });
     }
 
     // DEPRECATED - This code should be removed in v1.0.0.
     if (this.options.only && this.options.only.length > 0) {
       specifiedRules = this.options.only.map(toUpperCamelCase);
-      rules = defaultRules.filter(rule => {
+      rules = this.getAllRules().filter(rule => {
         return specifiedRules.indexOf(rule.name) >= 0;
       });
     }
 
     // DEPRECATED - This code should be removed in v1.0.0.
     if (this.options.except && this.options.except.length > 0) {
       specifiedRules = this.options.except.map(toUpperCamelCase);
-      rules = defaultRules.filter(rule => {
+      rules = this.getAllRules().filter(rule => {
         return specifiedRules.indexOf(rule.name) == -1;
       });
     }
 
     return rules;
   }
 
+  getAllRules() {
+    if (this.rules !== null) {
+      return this.rules;
+    }
+
+    this.rules = [];
+    this.rulePaths.map(rulePath => {
+      this.rules = this.rules.concat(this.loadRules(rulePath));
+    });
+
+    return this.rules;
+  }
+
+  loadRules(dir) {
+    var rules = [];
+
+    readdirSync(dir).forEach(file => {
```

Comment:
> The path can be a folder containing many rules (as with the default rules) or to a specific rule file.

I may have misunderstood what you meant, but if `dir` is the path to a rule file this seems to crash.

Example:

```
> fs.readdirSync("package.json")
Error: ENOTDIR: not a directory, scandir 'package.json'
    at Object.fs.readdirSync (fs.js:924:18)
```

I wonder if instead of using `readdirSync` we could use the same logic as we use to load the schema files:

https://github.com/cjoudrey/graphql-schema-linter/blob/d582c272aa082b2ee414e99f888921ea848f263b/src/configuration.js#L46-L60

Reply:
Hmm, not sure why there isn't coverage for the single file case-I'll look into that. 

To your second comment, consistent path options would be 💯 , I'll look into leveraging the same glob magic 👍

Reply:
K that seemed to sync up nicely in the latest commit.